test.js

```diff
@@ -480,6 +480,42 @@ describe('url-parse', function () {
       assume(data.href).equals('http://yahoo.com:808/?foo=bar');
     });
 
+    it('updates the port when updating host (IPv6)', function () {
+      var data = parse('http://google.com/?foo=bar');
+
+      assume(data.set('host', '[56h7::1]:808')).equals(data);
+
+      assume(data.hostname).equals('[56h7::1]');
+      assume(data.host).equals('[56h7::1]:808');
+      assume(data.port).equals('808');
+
+      assume(data.href).equals('http://[56h7::1]:808/?foo=bar');
+    });
+
+    it.skip('unsets the port when port is missing (IPv6)', function () {
+      var data = parse('http://google.com/?foo=bar');
+
+      assume(data.set('host', '56h7::1')).equals(data);
```

Comment:
The square brackets are required also when the port is missing.

- `http://1080::8:800:200C:417A/foo` is not a valid url.
- `http://[1080::8:800:200C:417A]/foo` is a valid url.